pyomo/contrib/incidence_analysis/interface.py

```diff
@@ -346,3 +346,44 @@ def dulmage_mendelsohn(self, variables=None, constraints=None):
         # Switch the order of the maps here to match the method call.
         # Hopefully this does not get too confusing...
         return var_partition, con_partition
+
+    def remove(self, variables, constraints):
+        """
+        Removes the specified variables and constraints (columns and
+        rows) from the cached incidence matrix. This is a "projection"
+        of the variable and constraint vectors, rather than something
+        like a vertex elimination.
+
+        Arguments:
+        ----------
+        variables: List
+            VarData objects whose columns will be removed from the
+            incidence matrix.
+        constraints: List
+            ConstraintData objects whose rows will be removed from the
+            incidence matrix.
+
+        """
+        if self.cached is IncidenceMatrixType.NONE:
+            raise RuntimeError(
+                "Attempting to remove variables and constraints from cached "
+                "incidence matrix,\nbut no incidence matrix has been cached."
+            )
+        to_exclude = ComponentSet(variables + constraints)
```

Comment:
This assumes that `variables` and `constraints` are "addable" sequences.  If one is a tuple and one is a list, this will error.  If either is a generator, then it will error.  Consider:
```python
to_exclude = ComponentSet(variables)
to_exclude.update(constraints)
```